index.js

```diff
@@ -33,29 +35,31 @@ function fastifyWebsocket (fastify, opts, next) {
         throw new Error('websocket handler can only be declared in GET method')
       }
 
-      let wsHandler = routeOptions.wsHandler
-      let handler = routeOptions.handler
+      if (routeOptions.path !== routeOptions.prefix) {
```

Comment:
This is solving, but what happens if the user register?

```
  fastify.register(
    function (instance, opts, next) {
      instance.get('/foo', { websocket: true }, ...)
      })
      next()
    },
    { prefix: '/foo' }
  )
```

Reply:
It works fine. because `('/foo/foo' !== '/foo')`

Reply:
had the time to check, I didn't realize the composition of that var 👍